src/backend/columnar/cstore_metadata_tables.c

```diff
@@ -41,28 +42,49 @@
 #include "utils/memutils.h"
 #include "utils/lsyscache.h"
 #include "utils/rel.h"
+#include "utils/relfilenodemap.h"
+
+
+/*
+ * Content of the first page in main fork, which stores metadata at file
+ * level.
+ */
+typedef struct ColumnarMetapage
+{
+	/*
+	 * Store version of file format used, so we can detect files from
+	 * previous versions if we change file format.
+	 */
+	int versionMajor;
+	int versionMinor;
```

Comment:
Thanks! I like the idea of starting the page with the version

Reply:
I feel like we are going somewhat down the road of a hybrid between the metadata tables approach and the metadata files approach. I agree, but I think we should have a clear boundary between the type of information stored in the metadata tables and the metapage.

One possibility is that table metadata goes in the metapage, and stripe/chunk metadata goes in metadata tables. But I'm not sure that's the right split.